capstone/src/main/java/com/google/sps/servlets/APKUploadServlet.java

```diff
@@ -72,14 +74,28 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
 
       // The line below retrieves the contents of the files 
       // in their bytes form for easy upload.
-      apk_file = IOUtils.toByteArray(file.getInputStream());
 
-      // The block of code below uploads only APK files to cloud storage.
+      // The block of code below uploads only APK files to cloud storage in chunks.
       blobId = BlobId.of(BUCKETNAME, fileName);
       blobInfo = BlobInfo.newBuilder(blobId).build();
-      storage.create(blobInfo, apk_file);
 
-    }
+      // The snippet below upload files to cloud storage bits by bits. This
+      // is essential to allow the upload of larger files through the use of
+      // a write channel.
+      try (WriteChannel writer = storage.writer(blobInfo)) {
+
+        apk_file = new byte[10_240];
```

Comment:
I would typically make this a variable in the class to:

a) make it obvious what this magical number is doing / means / why it was chosen
b) make it easy for anyone who needs to to come in and know what variable to change to increase or decrease chunk size